astrogrism/simulate/simulate.py

```diff
@@ -0,0 +1,84 @@
+from os import environ
+from pathlib import Path
+from tarfile import open as tar_open
+from tempfile import mkdtemp
+from warnings import warn
+
+from astropy.utils.data import download_file
+
+
+def generate_simulation_spectrum(grism, detector=None, temp_path=mkdtemp()):
+    """
+    Initializes and uses STSynphot to generate a Vega spectrum within the bandpass of a given grism
+
+    Parameters
+    ----------
+    grism : str
+        String representation of one of the four supported HST Grisms
+        Valid grisms: G141, G102, G280, G800L
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+
+    temp_path : str
+        Path to download necessary files for STSynphot. Fallsback to Python's
+        default temporary folder location
+
+    """
+    SIM_DATA_DIR = Path(temp_path) / "astrogrism_simulation_files"
+
+    if detector not in (1, 2, None):
+        raise ValueError("Invalid detector argument. Please choose 1 or 2")
+
+    # Prepare environment required for stsynphot to be imported
+    environ['PYSYN_CDBS'] = str(SIM_DATA_DIR / 'grp' / 'redcat' / 'trds')
+
+    # Always download our data files fresh. Ensures changes from redcat team will be picked up
+    # Download HST Instrument Data Files
+    SIM_DATA_DIR.mkdir(parents=True, exist_ok=True)
+    hst_data_files_archive = Path(download_file('https://ssb.stsci.edu/trds/tarfiles/synphot1.tar.gz')) # noqa
+    with tar_open(hst_data_files_archive) as tar:
+        tar.extractall(path=SIM_DATA_DIR)
+    # Download Vega CALSPEC Reference Atlas
+    calspec_dir = SIM_DATA_DIR / 'grp' / 'redcat' / 'trds' / 'calspec'
+    calspec_dir.mkdir(parents=True, exist_ok=True)
+    vega_reference_atlas = Path(download_file('https://archive.stsci.edu/hlsps/reference-atlases/cdbs/current_calspec/alpha_lyr_stis_010.fits')) # noqa
+    vega_reference_atlas.replace(calspec_dir / 'alpha_lyr_stis_010.fits')
+
+    # Now that we have all our reference files, we can import stsynphot
+    # (This is why it's not a top-line import)
+    from stsynphot import Vega, band # noqa
+    if grism == 'G141':
+        if detector:
+            warn("WFC3's G141 grism does not have multiple detectors. Ignoring detector argument",
```

Comment:
I don't know how I feel about warning here. Another option is to document that `detector` will be silently ignore in these cases in the docstring and not emit any warning.

Reply:
My preference of a warning over a silent ignore here is this warning should only fire when the user specifies a chip with instruments that don't have them. In this situation, I feel the user would be confused about what they're doing, so the warning gives them that heads up. Thoughts?

Reply:
I can buy that compromise @rosteen! Implemented